src/qt/walletmodel.cpp

```diff
@@ -79,6 +79,10 @@ void WalletModel::updateStatus()
 
 void WalletModel::pollBalanceChanged()
 {
+    // Avoid recomputing wallet balances unless a TransactionChanged or
+    // BlockTip notification was received.
+    if (!fForceCheckBalanceChanged && cachedNumBlocks == m_client_model.getNumBlocks()) return;
```

Comment:
Maybe this should be `>=`? Can't `tryGetBalances` return a block ahead of `m_client_model.getNumBlocks()`?

Alternatively drop 2nd arg from `tryGetBalances` and use `numBlocks = m_client_model.getNumBlocks()`?

Reply:
> Maybe this should be `>=`? Can't `tryGetBalances` return a block ahead of `m_client_model.getNumBlocks()`?

Not sure what goal of this suggestion is. >= would make the check less sensitive in case of a rollback. This code is already not sensitive enough because it is assuming chains the same height are identical.

> Alternatively drop 2nd arg from `tryGetBalances` and use `numBlocks = m_client_model.getNumBlocks()`?

Again not sure reason for this suggestion. It would make the PR no longer a direct revert of 0933a37078e1ce3a3d70983c3e7f4b3ac6c3fa37 and also no longer be correct when combined with #17954 and when the node tip runs ahead of the wallet last block processed.

Goal of this PR is just to make a one-line change that prevents unnecessary cross-process calls and allows #18160 to be reverted. There's room for improvement on top of PR, including removing the polling logic altogether, but by itself this PR should be an improvement on a few fronts

Reply:
> This code is already not sensitive enough because it is assuming chains the same height are identical

Indeed! 🤦